language-support/ts/daml-types/index.ts

```diff
@@ -93,10 +108,25 @@ const InterfaceBrand: unique symbol = Symbol();
  */
 export type Interface<IfId> = { readonly [InterfaceBrand]: IfId };
 
+/**
+ * Interface for objects representing Daml interfaces.
+ */
+export type InterfaceCompanion<
+  T extends object,
+  K,
+  I extends string = string,
+> = ContractTypeCompanion<T, K, I>;
+
+export type TemplateOrInterface<
+  T extends object,
+  K = unknown,
+  I extends string = string,
+> = Template<T, K, I> | InterfaceCompanion<T, K, I>;
```

Comment:
Admittedly, this is `= ContractTypeCompanion` right now. But as soon as `InterfaceCompanion` acquires a single property, it will become significantly but subtly different. And I do not think we will successfully retroactively draw that distinction in the various use sites later, if we do not draw the distinction now.

So _nominally_ `TemplateOrInterface` is the more specific choice, and a more conservative widening from `Template` for the various existing functions than we would other be making.